File: src/main/java/com/chatchatabc/jpademojava/application/rest/jwt/JwtService.java
package com.chatchatabc.jpademojava.application.rest.jwt;

import com.chatchatabc.jpademojava.domain.model.User;
import org.springframework.stereotype.Service;

@Service
public interface JwtService {

    /**
     * Generate JWT token
     *
     * @param user
     * @return
     */
    String generateToken(User user);

    /**
     * Validate token and get user id
     *
     * @param token
     * @return
     */
    String validateTokenAndGetId(String token);
}

File: src/main/kotlin/com/chatchatabc/jpademo/application/rest/jwt/JwtService.kt
package com.chatchatabc.jpademo.application.rest.jwt

import com.chatchatabc.jpademo.domain.model.User
import org.springframework.stereotype.Service

@Service
interface JwtService {
    /**
     * Generate JWT Token
     */
    fun generateToken(user: User): String;

    /**
     * Validate JWT Token and get ID
     */
    fun validateTokenAndGetId(token: String): String;
}